scripts/hid_configurator/gui/gui.py

```diff
@@ -0,0 +1,159 @@
+                print("Add {} to dropdown".format(device['product_id']))
+                self.add_dropdown_button(device['product_id'])
+
+    def initialize_device(self, pid):
+        self.device = Device.create_instance(DEVICE_TYPE[pid])
+        print("Initialize device with pid: {}".format(pid))
+
+    def show_possible_settings(self):
+        label = self.root.ids.settings_label
+        if self.device.pid == DEVICE_PID['keyboard']:
+            label.text = 'there are no available keyboard options for now'
+            self.add_dfu_buttons()
+        elif self.device.pid == DEVICE_PID['gaming_mouse']:
+            label.text = 'mouse options'
+            self.add_mouse_settings()
+            self.add_dfu_buttons()
+        else:
+            label.text = 'Unrecognised device'
+
+    def add_dropdown_button(self, pid):
+        dropdown = self.root.ids.dropdown
+        button = Button()
+        button.size_hint_y = None
+        button.height = '48dp'
+        button.text = DEVICE_TYPE[pid]
+        button.bind(on_release=lambda btn: self.show_fwinfo())
+        button.bind(on_release=lambda btn: self.show_possible_settings())
+        button.bind(on_release=lambda btn: dropdown.select(btn.text))
+        button.bind(on_release=lambda btn: self.clear_possible_settings())
+        button.bind(on_release=lambda btn: self.initialize_device(pid))
+        dropdown.add_widget(button)
+
+    def clear_dropdown(self):
+        dropdown = self.root.ids.dropdown
+        dropdown.clear_widgets()
+        grid_layout = GridLayout()
+        dropdown.add_widget(grid_layout)
+
+    def update_device_list(self):
+        self.clear_dropdown()
+        self.initialize_dropdown()
+
+    def show_fwinfo(self):
+        label = self.root.ids.fwinfo_label
+        returned = self.device.perform_fwinfo()
+        if returned:
+            flash_area_id, image_len, ver_major, ver_minor, ver_rev, ver_build_nr = returned
+            label.text = ('Firmware info\n'
+                          ' FLASH area id: {}\n'
+                          ' Image length: {}\n'
+                          ' Version: {}.{}.{}.{}').format(flash_area_id,
+                                                          image_len,
+                                                          ver_major,
+                                                          ver_minor,
+                                                          ver_rev,
+                                                          ver_build_nr)
+        else:
+            label.text = 'FW info request failed'
+
+    def show_load_list(self):
+        content = LoadDialog(load=self.load_list, cancel=self.dismiss_popup)
+        self._popup = Popup(title="Load a file list", content=content, size_hint=(0.9, 0.9))
+        self._popup.open()
+
+    def load_list(self, path, files_path):
+        print(path)
+        print(files_path)
```

Comment:
check if only one file is chosen

Reply:
FileChooserListView choose one file by default.